pages/blog/_posts/banger.mdx

```diff
@@ -0,0 +1,279 @@
+---
+heading: A Deep Dive into a Video Rendering Pipeline
+showSubtitle: true
+subtitle: banger.show is a video app maker that heavily relies on background data processing
+image: /assets/blog/banger/banger-main.jpg
+date: 2024-05-01
+author: Igor Samokhovets
+disableCTA: true
+---
+
+import YouTube from 'react-youtube-embed';
+
+Hi everyone! My name is [Igor Samokhovets](https://twitter.com/IgorSamokhovets) and I'm a music producer who goes by the artist name Tequila Funk. Together with Mark Beziaev we created [banger.show](https://banger.show/) -- a video maker app for musicians, DJs, and labels. It allows music industry people to create stunning visual assets for their music.
+
+<YouTube id="eSLKm6IfZM4" height={300}/>
+
+ Creating a video for your new song takes only a few minutes and you don't need to know any complex software because banger.show works in your browser.
+
+
+## Making background processing snappy
+
+At banger.show, we do a lot of background processing. Managing render states, generating visual assets in the background, and even handling the rendering process on remote distributed workers. We chose Inngest because there's no better way to handle background jobs if you're using Next.js.
+
+Let's dive into some parts of the video rendering pipeline.
```

Comment:
It would be nice for a little context, like "this pipeline receives an uploaded mp3 from the user and combines the audio with a generated visualization designed to match the audio to create a unique video for the user's music"

Reply:
Another thing we could add is "why use steps" for this - basically saying, that this allows it to run on serverless or sometimes things fail, etc. Are we using concurrency settings for this at all?

Reply:
@sylwiavargas I'm not sure how to put it simply:

- The pipeline receives the project created by the user
- Converts the audio for the more efficient rendering
- Controlling the flow of the render: sending user project to the render machine, updating statuses and progress, handling error retries and "stalled render" cases
- Waiting for the render to finish and perform post-render tasks like invalidating CDN cache, creating video thumbnails, sending email to the user when the video is ready